### ISSUE TITLE
in some (most) cases `blobs delete blob <hash>` seems to do nothing

### ISSUE THREAD

**Issue by divagant-martian** [n0-computer/iroh#2657]
Using main at [c70caaf](https://github.com/n0-computer/iroh/commit/c70caaf48ac39fcfb3b971366a76bff37b493ee9)

I think I have seen this work once, but in most cases It looks like it isn't doing anything.

### Reproduction steps
1. start iroh
  ```fish
  env IROH_DATA_DIR=/tmp/fresh_dir iroh start
  ```
2. start the console somewhere else
  ```fish
  env IROH_DATA_DIR=/tmp/fresh_dir iroh console
  ```
3. add a blob
  ```fish
  > blobs list blobs # empty, dir is fresh
  > blobs add CONTRIBUTING.md
  Adding CONTRIBUTING.md as /home/divagant-martian/Workspace/iroh/CONTRIBUTING.md...
  - /home/divagant-martian/Workspace/iroh/CONTRIBUTING.md: 5.20 KiB pnzfqspcskqur4kuwck7hc7yxkkm4exuhle5urbssozi3zttjdjq
Total: 5.20 KiB

Blob: pnzfqspcskqur4kuwck7hc7yxkkm4exuhle5urbssozi3zttjdjq
All-in-one ticket: blobacm6x2wx23oc2ssbxqubxlaa2jeptmvbjbjewygviwqx3usz3f47oaiqnb2hi4dthixs6ztpn4xgeylsf4babqfiaah4ivyazhmaaauk4abaa63slbe6fevbjdyvjmev6of7rouuzyjpiowj3jcdfe5srxthgsgt
> blobs list blobs
 pnzfqspcskqur4kuwck7hc7yxkkm4exuhle5urbssozi3zttjdjq (5.20 KiB)
  ```
5. "delete" the blob
```fish
> blobs delete blob pnzfqspcskqur4kuwck7hc7yxkkm4exuhle5urbssozi3zttjdjq
# empty response, so it's assumed all worked
> blobs list blobs
 pnzfqspcskqur4kuwck7hc7yxkkm4exuhle5urbssozi3zttjdjq (5.20 KiB)
# wut face
```

**Comment by b5** [comment]
Others have run into this, I've love to get to the bottom of a fix before the v0.24.0 release

**Comment by divagant-martian** [comment]
FYI can still reproduce basically 100% of the time today at 5bbcb6083
